Marlin/Marlin_main.cpp

```diff
@@ -1019,6 +1035,22 @@ inline void get_serial_commands() {
 
       // If command was e-stop process now
       if (strcmp(command, "M112") == 0) kill(PSTR(MSG_KILLED));
+      #if ENABLED(PROBE_FAIL_PANIC)
+        if (strcmp(command, "G26") == 0) {
+          LCD_MESSAGEPGM(WELCOME_MSG);
+          probe_fail = false;
+        }
+        else if(probe_fail && (strchr(command, 'G') != NULL || strchr(command, 'M') != NULL) && strstr(command, "M105") == NULL) {
+          for(uint8_t j=0; j < MAX_CMD_SIZE; j++)
+            command_queue[cmd_queue_index_r][j] = 0;
+          if(commands_in_queue) {
+            commands_in_queue = 1;
+            cmd_queue_index_r = (cmd_queue_index_r + 1) % BUFSIZE;
+          }
+          serial_count = 0;
+          return;
+        }
+      #endif
```

Comment:
I see! So, I'm also wondering why we need a new command or a "special" kind of panic, since we have `M999` (and recently added `M999 S1` to preserve the `rx_buffer`). Also, if, as you say, the buffer is flushed then there should be no need to do special handling of `G26` outside of the normal command buffer. (We want to minimize calls to `strcmp` in the gcode parser.)